/*
 * This file was automatically generated by EvoSuite
 * Sat May 24 19:46:19 GMT 2025
 */

package original;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import original.Solution3069;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false) 
public class Solution3069_ESTest extends Solution3069_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Solution3069 solution3069_0 = new Solution3069();
      int[] intArray0 = new int[6];
      intArray0[1] = (-243);
      intArray0[3] = (-4077);
      int[] intArray1 = solution3069_0.resultArray(intArray0);
      assertArrayEquals(new int[] {0, (-243), 0, (-4077), 0, 0}, intArray0);
      assertArrayEquals(new int[] {0, 0, (-4077), (-243), 0, 0}, intArray1);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Solution3069 solution3069_0 = new Solution3069();
      // Undeclared exception!
      try { 
        solution3069_0.resultArray((int[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("original.Solution3069", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Solution3069 solution3069_0 = new Solution3069();
      int[] intArray0 = new int[0];
      // Undeclared exception!
      try { 
        solution3069_0.resultArray(intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // Index 0 out of bounds for length 0
         //
         verifyException("original.Solution3069", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      Solution3069 solution3069_0 = new Solution3069();
      int[] intArray0 = new int[12];
      intArray0[0] = 381;
      int[] intArray1 = solution3069_0.resultArray(intArray0);
      assertEquals(12, intArray1.length);
      assertNotSame(intArray1, intArray0);
  }
}
